rviz_default_plugins/src/rviz_default_plugins/displays/laser_scan/laser_scan_display.cpp

```diff
@@ -50,29 +50,28 @@ LaserScanDisplay::LaserScanDisplay()
 : point_cloud_common_(std::make_unique<rviz_default_plugins::PointCloudCommon>(this)),
   queue_size_property_(std::make_unique<rviz_common::QueueSizeProperty>(this, 10)),
   projector_(std::make_unique<laser_geometry::LaserProjection>()),
+  filter_tolerance_(0, 0),
   transformer_guard_(
     std::make_unique<rviz_default_plugins::transformation::TransformerGuard<
       rviz_default_plugins::transformation::TFFrameTransformer>>(this, "TF"))
 {}
 
 void LaserScanDisplay::onInitialize()
 {
-  RTDClass::onInitialize();
+  MFDClass::onInitialize();
   point_cloud_common_->initialize(context_, scene_node_);
   transformer_guard_->initialize(context_);
 }
 
 void LaserScanDisplay::processMessage(sensor_msgs::msg::LaserScan::ConstSharedPtr scan)
 {
-  // TODO(Martin-Idel-SI): Reenable once tf_filter is ported or delete if necessary
 //  Compute tolerance necessary for this scan
-//  ros::Duration tolerance(scan->time_increment * scan->ranges.size());
-//  if (tolerance > filter_tolerance_)
-//  {
-//    filter_tolerance_ = tolerance;
-//    tf_filter_->setTolerance(filter_tolerance_);
-//  }
-
+  rclcpp::Duration tolerance(static_cast<int32_t>(static_cast<rcl_duration_value_t>(
+      scan->time_increment * scan->ranges.size())), 0);
+  if (tolerance > filter_tolerance_) {
+    filter_tolerance_ = tolerance;
+    tf_filter_->setTolerance(filter_tolerance_);
```

Comment:
I'm having a problem rendering laser scans from Gazebo. I get the following message repeated:

```
[INFO] [rviz]: [signalFailure] Drop message: frame 'base_link' at time 2043.937 for reason(0)
[INFO] [rviz]: [signalFailure] Drop message: frame 'base_link' at time 2044.037 for reason(0)
[INFO] [rviz]: [signalFailure] Drop message: frame 'base_link' at time 2044.138 for reason(0)
...
```

Which is [coming from `tf2_ros::MessageFilter`](https://github.com/ros2/geometry2/blob/fccfd986ad82208dc64d0f1d650dc87cf336cae7/tf2_ros/include/tf2_ros/message_filter.h#L652). Specifically, I've traced the error message to [this line](https://github.com/ros2/geometry2/blob/fccfd986ad82208dc64d0f1d650dc87cf336cae7/tf2_ros/include/tf2_ros/message_filter.h#L539).

I probably introduced the bug in https://github.com/ros2/geometry2/pull/121. I think transform requests are timing out immediately due to a [zero timeout duration](https://github.com/ros2/geometry2/blob/fccfd986ad82208dc64d0f1d650dc87cf336cae7/tf2_ros/include/tf2_ros/message_filter.h#L327).

Aside from this bug, I'm not sure how the message filter worked properly in RViz before, since if the tolerance defaults to zero I wouldn't expect `processMessage` to be called at all. I'll continue digging and see if I can resolve the issue before moving forward with this PR.

Reply:
Can you pls tell how to reproduce this. I was getting this error only when the fixed frame of rviz was not correct

Reply:
I tried running this [confbot_robot simulation tutorial](https://github.com/Karsten1987/confbot_robot/blob/master/tutorial/Linux_Tutorial.md#simulating-the-robot-in-gazebo) and when I change the fixed frame to the `odom` frame, I get the reported dropped messages and sensors stop rendering in RViz (laser and camera).

Reply:
I think this should fix the issue: https://github.com/ros2/geometry2/pull/146